### PR TITLE
IPEHR-167: ADL14 upload, initial

### PR DESCRIPTION
**Pull request by Setest**

### CODE REVIEW

File: src/pkg/docs/parser/adl14/parser_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	"testing"
+)
+
+func TestParser_Validate(t *testing.T) {
+	p := NewParser()
+
+	t.Run("Invalid XML", func(t *testing.T) {
+		r := p.Validate([]byte(`<xml>invalid>/`), model.ADLTypeXML)
+
+		if r {
```
Comment (SealTV) [inline src/pkg/docs/parser/adl14/parser_test.go]: you can write like this
```go
if p.Valudate([]byte(`<xml>invalid/`)) {
    ....
}
```

File: src/pkg/docs/parser/adl14/parser_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+	t.Run("Valid XML", func(t *testing.T) {
+		r := p.Validate([]byte(`<xml></xml>`), model.ADLTypeXML)
+
+		if !r {
+			t.Errorf("adl14.Validate() xml is invalid")
+			return
+		}
+	})
+}
```
Comment (SealTV) [inline src/pkg/docs/parser/adl14/parser_test.go:33]: dAnd in total we can write here table test
```go 
tests := struct{
    name string
    xmlStr string
    want bool
}{
   {
      "1. Invalid XML",
       `<xml>invalid/`,
       false,
   },
   {
    "2. Valid XML",
    `<xml>val</xml>`,
    true,
   },
}
for _ , tt := range tests {
    t.Run(tt.name, func(t *testing.T){
        parser := NewParser()
        if parser.Validate([]byte(tt.xmlStr, model.ADLTypeXML)) != tt.want {
            t.Error("adl14.Validate() error")
        }
    })
}
```

**Review by SealTV — APPROVED** [review]

File: src/pkg/api/template_test.go [hunk 3]
```diff
@@ -148,7 +149,149 @@ func TestTemplateHandler_GetByID(t *testing.T) {
...
+			// Mock for auth user service
+			userSvc := mocks.NewMockUserService(ctrl)
+			userSvc.EXPECT().VerifyAccess(gomock.Any(), gomock.Any()).Return(nil)
+
+			api := API{
+				Template: NewTemplateHandler(tSvc, ""),
+				User:     NewUserHandler(userSvc),
+			}
+
+			router := api.setupRouter(api.buildDefinitionAPI())
```
Comment (1db1) [inline src/pkg/api/template_test.go]: We can take API building outside the loop. 
Then it won't build every time and the log will be cleaner

Comment (SealTV) [reply src/pkg/api/template_test.go]: It does not take a long time.  And if you set up mocks outside for loop, it would affect other test cases.

**Review by 1db1 — APPROVED** [review]

**Pull request merged**